internal/circuit/mapfunc_test.go

```diff
@@ -0,0 +1,103 @@
+package circuit
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/crossplane/crossplane/v2/internal/metrics"
+)
+
+type fakeBreaker struct {
+	state             State
+	recordAllowedCall int
+}
+
+func (f *fakeBreaker) GetState(context.Context, types.NamespacedName) State {
+	return f.state
+}
+
+func (f *fakeBreaker) RecordEvent(context.Context, types.NamespacedName, EventSource) {}
+
+func (f *fakeBreaker) RecordAllowed(context.Context, types.NamespacedName) {
+	f.recordAllowedCall++
+}
+
+func TestNewMapFuncMetrics(t *testing.T) {
```

Comment:
FWIW we don't unit test any of our other metrics implementations. I'm comfortable leaving this out. If you do want to keep it, it should use `cmp` for comparisons.